packages/hops/lib/runtime.d.ts

```diff
@@ -0,0 +1,83 @@
+import { BrowserRouterProps } from 'react-router-dom';
+
+declare module 'hops' {
+  type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+
+  export class Miss extends React.Component<any, any> {}
+
+  export interface StatusProps {
+    code?: number;
+  }
+  export class Status extends React.Component<StatusProps, any> {}
+
+  export interface HeaderProps {
+    name: string;
+    value: string;
+  }
+
+  export class Header extends React.Component<HeaderProps, any> {}
+
+  export interface ImportComponentRenderOptions<Props> {
+    Component: React.ComponentType<Props>;
+    error: boolean;
+    loading: boolean;
+  }
+
+  export interface ImportComponentOptions<Props> {
+    loader?: (load: Promise<any>) => Promise<any>;
+    render?: (
+      options: ImportComponentRenderOptions<Props> & Props
+    ) => React.ReactNode;
+  }
+
+  export function importComponent<Props, Exports>(
+    load: () => Promise<{ default: React.ComponentType<Props> }>
+  ): React.ComponentType<ImportComponentOptions<Props> & Props>;
+
+  export function importComponent<Props, Exports>(
+    load: () => Promise<Exports>,
+    resolver: (exports: Exports) => React.ComponentType<Props>
+  ): React.ComponentType<ImportComponentOptions<Props> & Props>;
+
+  export function importComponent<Props>(
+    module: string,
+    exportName?: string
+  ): React.ComponentType<ImportComponentOptions<Props>>;
+
+  export interface ServerData {
+    [key: string]: any;
+  }
+
+  export interface Config {
+    [key: string]: any;
+  }
+
+  export const ServerDataContext: React.Context<ServerData>;
+
+  export const ConfigContext: React.Context<Config>;
+
+  export function withServerData<
```

Comment:
This looks quite complicated. Wouldn't it do the trick to augment a component of type `React.ComponentType<P>` to `React.ComponentType<P & C>`?

Reply:
I don't think so, because you are not supposed to pass in `serverData` as a prop yourself, but it is a HoC, therefore it should be available in the wrapped component, but should not be marked as a required prop for you to pass into the component.